### PR TITLE
Update IdP Overview Page

### PR DESCRIPTION
**Pull request by alexfornuto**
## Summary

- Removes Dex/FreeIPA content (saved in non-indexed page, for posterity and/or to be revived later).
- Updates the IdP Overview page to prepare users for what to expect when following the IdP-specific docs

## Related issues

Closes https://github.com/pomerium/docs/issues/9
Closes https://github.com/pomerium/docs/issues/16

## Checklist

- [x] reference any related issues
- [x] updated docs
~updated unit tests~
~updated UPGRADING.md~
- [x] add appropriate tag (`improvement` / `bug` / etc)
- [x] ready for review

### CODE REVIEW

**Comment by alexfornuto** [comment]
@desimone what else would you like to see covered in this page?

File: docs/docs/identity-providers/readme.md [hunk 1]
```diff
@@ -8,23 +8,14 @@ description: >-
 
 # Identity Provider Configuration
 
-This article describes how to configure Pomerium to use a third-party identity service for single-sign-on.
+Pomerium helps provide a zero-trust security model by verifying that every request to your upstream applications are from an authenticated and authorized user. User and group information is sourced from your Identity Provider (**IdP**). Pomerium communicates with IdPs following the [OpenID Connect][openid connect] (**OIDC**) standard for user identity, and a service account and/or API token for group/directory information. The steps for integrating Pomerium with an IdP are specific to each provider, but they generally share the same base requirements:
```
Comment (desimone) [inline docs/docs/identity-providers/readme.md:11]:
```suggestion
Pomerium helps provide a zero-trust security model by verifying that every request to your upstream applications is from an authenticated and authorized user. User and group information is sourced from your Identity Provider (**IdP**). Pomerium communicates with IdPs following the [OpenID Connect][openid connect] (**OIDC**) standard for user identity, and a service account and/or API token for group/directory information. The steps for integrating Pomerium with an IdP are specific to each provider, but they generally share the same base requirements:
```

File: docs/docs/identity-providers/readme.md [hunk 2]
```diff
@@ -8,23 +8,14 @@ description: >-
...
 
-There are a few configuration steps required for identity provider integration. Most providers support [OpenID Connect] which provides a standardized identity and authentication interface.
+- A **[Redirect URL](https://www.oauth.com/oauth2-servers/redirect-uris/)** pointing back to Pomerium. For example, `https://${authenticate_service_url}/oauth2/callback`.
+- A **[Client ID]** and **[Client Secret]**.
+- A **[Service Account]** for additional IdP Data. This is enables Pomerium administrators to write policies around groups, or any other data that doesn't uniquely identify an end-user, as defined in the IdP.
```
Comment (desimone) [inline docs/docs/identity-providers/readme.md:15]:
```suggestion
- A **[Service Account]** for additional IdP Data. This enables Pomerium administrators to write policies around groups, or any other data that doesn't uniquely identify an end-user, as defined in the IdP.
```

Comment (desimone) [reply docs/docs/identity-providers/readme.md:15]: I'm not sure what
> or any other data that doesn't uniquely identify an end-user, as defined in the IdP
means

Could you elaborate?

File: docs/docs/identity-providers/readme.md (hunk 1 above)
Comment (desimone) [reply docs/docs/identity-providers/readme.md:11]: Pomerium provides single-sign-on authentication and user identity details by integrating with your downstream identity provider of choice. That authentication integration is achieved using OAuth2, and OIDC. Where available, Pomerium also supports pulling additional data (like groups) using directory synchronization. An additional API token is required for directory sync.

File: docs/docs/identity-providers/readme.md [hunk 3]
```diff
@@ -8,23 +8,14 @@ description: >-
 
 # Identity Provider Configuration
 
-This article describes how to configure Pomerium to use a third-party identity service for single-sign-on.
+Pomerium helps provide a zero-trust security model by verifying that every request to your upstream applications are from an authenticated and authorized user. User and group information is sourced from your Identity Provider (**IdP**). Pomerium communicates with IdPs following the [OpenID Connect][openid connect] (**OIDC**) standard for user identity, and a service account and/or API token for group/directory information. The steps for integrating Pomerium with an IdP are specific to each provider, but they generally share the same base requirements:
 
-There are a few configuration steps required for identity provider integration. Most providers support [OpenID Connect] which provides a standardized identity and authentication interface.
+- A **[Redirect URL](https://www.oauth.com/oauth2-servers/redirect-uris/)** pointing back to Pomerium. For example, `https://${authenticate_service_url}/oauth2/callback`.
```
Comment (desimone) [inline docs/docs/identity-providers/readme.md:13]:
```suggestion
- A **[Redirect URL](https://www.oauth.com/oauth2-servers/redirect-uris/)** pointing back to Pomerium. For example, `https://${authenticate_service_url}/oauth2/callback`.
```
It's worth highlighting that the path `/oauth2/callback` is hardcoded. And that this URL is set within the identity provider, not Pomerium.

File: docs/docs/identity-providers/readme.md (hunk 2 above)
Comment (alexfornuto) [reply docs/docs/identity-providers/readme.md:15]: I was just covering my bases for what I didn't yet know Pomerium could do. Sounds like I should remove it.

**Review by desimone — APPROVED** [review]

**Pull request merged**